src/lib/operators/sort.cpp

```diff
@@ -341,21 +341,17 @@ class Sort::SortImpl {
       std::stable_sort(_row_id_value_vector.begin(), _row_id_value_vector.end(),
                        [comparator](RowIDValuePair a, RowIDValuePair b) { return comparator(a.second, b.second); });
     };
-    if (_order_by_mode == OrderByMode::Ascending || _order_by_mode == OrderByMode::AscendingNullsLast) {
+    if (_order_by_mode == OrderByMode::Ascending) {
       sort_with_comparator(std::less<>{});
     } else {
       sort_with_comparator(std::greater<>{});
     }
 
     // 2b. Insert null rows if necessary
     if (!_null_value_rows.empty()) {
-      if (_order_by_mode == OrderByMode::AscendingNullsLast || _order_by_mode == OrderByMode::DescendingNullsLast) {
-        // NULLs last
-        _row_id_value_vector.insert(_row_id_value_vector.end(), _null_value_rows.begin(), _null_value_rows.end());
-      } else {
-        // NULLs first (default behavior)
-        _row_id_value_vector.insert(_row_id_value_vector.begin(), _null_value_rows.begin(), _null_value_rows.end());
-      }
+      // NULLs come before all values. The SQL standard allows for this to be implementation-defined. We used to have
```

Comment:
That's kind of a strange definition in my opinion. I'd vote for doing the same as HANA and MySQL (actually does not matter if we join the larger or smaller DB community, I just don't like the handling of HSQLDB) as shown on the site Jan linked: https://docs.mendix.com/refguide/null-ordering-behavior

Reply:
The advantage of the NULLS FIRST approach is simply that we need less code. Both in the sort and in code that handles sorted data (e.g., scan_sorted_segment).

Reply:
Hmm, I don't like it. But than again sorting with NULLs sucks anyways. So when you say it's easier, I trust you here and that weights more.

Reply:
Add a comment to types.hpp? I think it's rather hidden right now.